packages/fxa-auth-server/lib/payments/paypal-client.ts

```diff
@@ -358,4 +401,20 @@ export class PayPalClient {
     );
     return result.text;
   }
+
+  public async transactionSearch(
+    options: TransactionSearchOptions
+  ): Promise<NVPTransactionSearchResponse> {
+    const data = {
+      STARTDATE: toIsoString(options.startDate),
+      ENDDATE: options.endDate ? toIsoString(options.endDate) : undefined,
+      EMAIL: options.email,
+      INVNUM: options.invoice,
+      TRANSACTIONID: options.transactionId,
```

Comment:
huh I see in the type declaration above that only `startDate` is not optional, so I'm kinda surprised that the type checker didn't complain about lines 411-413.  (I don't have the branch open locally; I'm sure there's a good reason.)

Reply:
They'll just be undefined in that case. In TS you can still use optional types like this.